### ISSUE TITLE
Cant create private posts

### ISSUE THREAD

**Issue by Ixiter** [friendica/friendica#6579]
### Expected behavior
2019.01
frio
I chosse a group and create a post. The post should be a private post, visible only to the group members.

### Actual behavior
The post becomes public

### Steps to reproduce the problem
create a private post (maybe only on my host), check access rights, choose a group

### Friendica version you encountered the problem
2019.01 master

### Friendica source (git, zip)
git

### PHP version
7.3

### SQL version
5.7

### Additional information
I thought it could been  caused by a patch or addon., so I cloned an untouched master branch and made it the webroot of my host. Didnt help.
I found this jot_geotag.tpl bug and unchecked the "location from Browser" option. Didnt help.
I deactivated addons that might cause the behavior .. didnt help.
Additionally I tried with different themes. Didnt help.

**Comment by Ixiter** [comment]
Thank you. Yes the ACL looked exactly like that and its how I understood and used the permissions tab.
Whe i scroll down in that tab, the group members are highlighted green.

As you saw, my posts didnt show the lock symbol and werent private.

**Comment by MrPetovan** [comment]
Can anyone else reproduce?

**Comment by tobiasd** [comment]
I could not reproduce this with the latest development code (5d2fce789a4b29ff7e7cfb9f2b89968453094da7) sending out test postings using Firefox and Vivaldi on Devuan stable, frio and vier.

**Comment by MrPetovan** [comment]
From a reply in the forum thread: https://friendica.mrpetovan.com/redir/47?url=https://social.yl.ms/display/e18176ef-865c-580b-f413-67c954945353

It looks like the group-directed posts are behaving like private posts, they just aren't looking like private posts (no lock display).

**Comment by tobiasd** [comment]
The postings from @Ixiter got reaction by non-group members.

And for me the private postings are clearly marked as such with the lock displayed, even when switching to frio.

**Comment by MrPetovan** [comment]
@Ixiter Have you been able to reproduce it consistently? Same/Different groups?

**Comment by AlfredSK** [comment]
Concerning the 'lock'. Please note there is no lock when using frio on mobile devices.
Just sayin'. It's not related to this issue. Frio mobile doesn't show the lock on mobile devices.

**Comment by Ixiter** [comment]
Still running 2019.03-dev - 1300
Still cant post privately to groups. No matter which group, no matter which device i try.

When I send the post, this javascript error pops up in developer console:
```
jquery.min.js:4 Uncaught DOMException: Failed to execute 'createElement' on 'Document': The tag name provided ('2') is not a valid name.
    at Function.n.parseHTML (https://dev.socivitas.com/view/asset/jquery/dist/jquery.min.js:4:17703)
    at new n.fn.init (https://dev.socivitas.com/view/asset/jquery/dist/jquery.min.js:2:24572)
    at n (https://dev.socivitas.com/view/asset/jquery/dist/jquery.min.js:2:405)
    at ACL.on_submit (https://dev.socivitas.com/view/js/acl.js:89:2)
    at HTMLFormElement.dispatch (https://dev.socivitas.com/view/asset/jquery/dist/jquery.min.js:3:7537)
    at HTMLFormElement.r.handle (https://dev.socivitas.com/view/asset/jquery/dist/jquery.min.js:3:5620)
n.parseHTML @ jquery.min.js:4
n.fn.init @ jquery.min.js:2
n @ jquery.min.js:2
ACL.on_submit @ acl.js:89
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
```
In ACL.on_submit() it tries to add the groupIds with hidden fields. 
line 89 in acl.js is line 3 in the snippet!

```javascript
ACL.prototype.on_submit = function(){
	var aclfields = $("#acl-fields").html("");
	$(this.allow_gid).each(function(i,v){
		aclfields.append("<input type='hidden' name='group_allow[]' value='"+v+"'>");
	});
	$(this.allow_cid).each(function(i,v){
		aclfields.append("<input type='hidden' name='contact_allow[]' value='"+v+"'>");
	});
	$(this.deny_gid).each(function(i,v){
		aclfields.append("<input type='hidden' name='group_deny[]' value='"+v+"'>");
	});
	$(this.deny_cid).each(function(i,v){
		aclfields.append("<input type='hidden' name='contact_deny[]' value='"+v+"'>");
	});
};
```

**Comment by annando** [comment]
A posting to a non private group is always public.

**Comment by Ixiter** [comment]
I did `console.log(this.allow_gid)` and it showed "<10>" ;)
This looks a pretty wrong value for a value that should be eached ;)

**Comment by Ixiter** [comment]
> A posting to a non private group is always public.

?!? How to set a group public or private ?

**Comment by Ixiter** [comment]
I also cant change permissions when do a post. It also throws errors when I try to select show/hide on groups in the permissions tab.
```
VM11613 acl.js:166 Uncaught TypeError: this.allow_gid.remove is not a function
    at ACL.set_allow (VM11613 acl.js:166)
    at ACL.on_button_show (VM11613 acl.js:142)
    at HTMLDocument.dispatch (VM11604 jquery.min.js:3)
    at HTMLDocument.r.handle (VM11604 jquery.min.js:3)
ACL.set_allow @ VM11613 acl.js:166
ACL.on_button_show @ VM11613 acl.js:142
dispatch @ VM11604 jquery.min.js:3
r.handle @ VM11604 jquery.min.js:3
3VM11613 acl.js:164 Uncaught TypeError: this.allow_gid.push is not a function
    at ACL.set_allow (VM11613 acl.js:164)
    at ACL.on_button_show (VM11613 acl.js:142)
    at HTMLDocument.dispatch (VM11604 jquery.min.js:3)
    at HTMLDocument.r.handle (VM11604 jquery.min.js:3)
ACL.set_allow @ VM11613 acl.js:164
ACL.on_button_show @ VM11613 acl.js:142
dispatch @ VM11604 jquery.min.js:3
r.handle @ VM11604 jquery.min.js:3

```

**Comment by annando** [comment]
The group itself must be a private group. This is an option in the user configuration.

**Comment by Ixiter** [comment]
I dont get that. When I edit a group, I can change the name and add and remove members. Where can I set private/public?

**Comment by Ixiter** [comment]
When I click "Show all" and reselect the group, it works!
Then it shows `this.allow_gid` as an array with the groupID I selected.

### Steps to reproduce the problem
 1. Goto your network page
 1. select a group in the groups widget
 1. try to create a post. Group and groupmembers are preselected in the permimssions tab. Dont change this!
 1. open dev console, send the post and watch the dev console if it shows an error

**Comment by AlfredSK** [comment]
I have the impression you're both talking about different things. Groups vs. Forums.

**Comment by annando** [comment]
@AlfredSK örgs, you are right ... Possibly I shouldn't answer issues while debugging, coding and watching TV at the same time ...

**Comment by Ixiter** [comment]
> @AlfredSK örgs, you are right ... Possibly I shouldn't answer issues while debugging, coding and watching TV at the same time ...

örgs isnt an english word :D

Anyways .. I think its a bug in the permissions preselection thing, when others than me experience the same bug by following the steps I described in my  comment above

Please can you check.

**Comment by Ixiter** [comment]
Update: This bug is on all preselected areas on the /network page. When i seölect a forum,. and try to post, it is preselected in the permission, but you cannot change it, before you clicked "Visible for everybody" and the reselect the forum or whatever you want. 

Same with protocols. When I select ActivityPub and start a new post, only users that support AP are selected in the permissions. But they arent really. Same procedure .. you need to click "Visible for everybody" and reselect the users, to be sure t becomes a post to only these users.

In short ... *Private messaging is buggy, when you select a network area in the widgets and try to post into it.*

Please someone who knows where and how the preselect permissions thing is done, take a look on it and fix it. :)

**Comment by Ixiter** [comment]
Fixed it. Will PR tomorrow

**Comment by Ixiter** [comment]
PR #6638

**Comment by MrPetovan** [comment]
Fixed by #6638
